pkg/skaffold/schema/latest/config.go

```diff
@@ -1522,6 +1525,20 @@ type SyncHooks struct {
 	PostHooks []SyncHookItem `yaml:"after,omitempty"`
 }
 
+// RenderHookItem describes a single lifecycle hook to execute before or after each deployer step.
+type RenderHookItem struct {
+	// HostHook describes a single lifecycle hook to run on the host machine.
+	HostHook *HostHook `yaml:"host,omitempty" yamltags:"oneOf=render_hook"`
+}
+
+// RenderHooks describes the list of lifecycle hooks to execute before and after each render step.
+type RenderHooks struct {
+	// PreHooks describes the list of lifecycle hooks to execute *before* each render step. Container hooks will only run if the container exists from a previous deployment step (for instance the successive iterations of a dev-loop during `skaffold dev`).
+	PreHooks []RenderHookItem `yaml:"before,omitempty"`
+	// PostHooks describes the list of lifecycle hooks to execute *after* each render step.
+	PostHooks []RenderHookItem `yaml:"after,omitempty"`
+}
```

Comment:
you don't need `RenderHookItem`. You can just define:
```
type RenderHooks struct {
	// PreHooks describes the list of lifecycle hooks to execute *before* each render step. Container hooks will only run if the container exists from a previous deployment step (for instance the successive iterations of a dev-loop during `skaffold dev`).
	PreHooks []HostHook `yaml:"before,omitempty"`
	// PostHooks describes the list of lifecycle hooks to execute *after* each render step.
	PostHooks []HostHook `yaml:"after,omitempty"`
}
```
since `render` like `build` doesn't need `ContainerHook`s. See: https://github.com/GoogleContainerTools/skaffold/blob/0164731ae119370aedd71501eb5cc1dfa1ccb3fa/pkg/skaffold/schema/latest/config.go#L1502-L1507

Reply:
After talking w/ Cloud Deploy team, they mentioned they will likely need to add support for another Hook type that runs in a container (vs on the Host) at a later date.  As such I think it makes sense to leave this indirection for now as we will be adding more hook types for render in the future.